### PR TITLE
Remove StepControllers

### PR DESCRIPTION
**Pull request by wthrowe**
The only one that was usable was BinaryFraction, so just always use that implementation.  SimpleTimes and SplitRemaining always quickly crashed by overflowing the Rational internals.  FullSlab is easy to duplicate by choosing a larger step size than slab size.

### Upgrade instructions

Remove any "StepController" options from your input files.  The "BinaryFraction" value is now used unconditionally.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Time/ChooseLtsStepSize.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+
+TimeDelta choose_lts_step_size(const Time& time, const double desired_step) {
+  ASSERT((time.fraction().denominator() &
+          (time.fraction().denominator() - 1)) == 0,
+         "Not at a binary-fraction time within slab: " << time.fraction());
+  const TimeDelta full_slab =
+      desired_step > 0 ? time.slab().duration() : -time.slab().duration();
```
Comment (kidder) [inline src/Time/ChooseLtsStepSize.cpp]: 0.0

File: src/Time/ChooseLtsStepSize.hpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Time/Time.hpp"
```
Comment (kidder) [inline src/Time/ChooseLtsStepSize.hpp]: could forward declare

**Review by kidder — COMMENTED** [review]
optional changes for you to consider

**Comment by wthrowe** [comment]
Did both, and also stopped running the assertion test in release mode.

**Review by kidder — APPROVED** [review]

**Pull request merged**